hazelcast-client/src/test/java/com/hazelcast/client/config/XmlClientConfigPreProcessorTest.java

```diff
@@ -0,0 +1,257 @@
+        String xml = "<hazelcast-client>\n" +
+                "   <network>" +
+                "        <import resource=\"\"/>\n" +
+                "   </network>" +
+                "</hazelcast-client>";
+
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
+        XmlClientConfigBuilder configBuilder = new XmlClientConfigBuilder(bis);
+        ClientConfig config = configBuilder.build();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testHazelcastElementOnlyAppearsOnce() throws Exception {
+        String xml = "<hazelcast-client>\n" +
+                "   <hazelcast-client>" +
+                "   </hazelcast-client>" +
+                "</hazelcast-client>";
+
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
+        XmlClientConfigBuilder configBuilder = new XmlClientConfigBuilder(bis);
+        ClientConfig config = configBuilder.build();
+    }
+
+    @Test
+    public void readVariables() {
+        String xml =
+                "<hazelcast-client>\n" +
+                        "<executor-pool-size>${executor.pool.size}</executor-pool-size>" +
+                        "</hazelcast-client>";
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
+        XmlClientConfigBuilder configBuilder = new XmlClientConfigBuilder(bis);
+
+        Properties properties = new Properties();
+        properties.setProperty("executor.pool.size", "40");
+        configBuilder.setProperties(properties);
+
+        ClientConfig config = configBuilder.build();
+        assertEquals(40, config.getExecutorPoolSize());
+    }
+
+    @Test
+    public void testImportConfigFromResourceVariables() throws IOException {
+        File file = createConfigFile("foo", "bar");
+        FileOutputStream os = new FileOutputStream(file);
+        String networkConfig = "<hazelcast-client>" +
+                "<network>" +
+                "<cluster-members>" +
+                "<address>192.168.100.100</address>" +
+                "<address>127.0.0.10</address>" +
+                "</cluster-members>" +
+                "<smart-routing>false</smart-routing>" +
+                "<redo-operation>true</redo-operation>" +
+
+                "<socket-interceptor enabled=\"true\">" +
+                "<class-name>com.hazelcast.examples.MySocketInterceptor</class-name>" +
+                "<properties>" +
+                "<property name=\"foo\">bar</property>" +
+                "</properties>" +
+                "</socket-interceptor>" +
+                "        </network>" +
+                "</hazelcast-client>";
+        writeStringToStreamAndClose(os, networkConfig);
+
+        String xml = "<hazelcast-client>\n" +
+                "    <import resource=\"${config.location}\"/>\n" +
+                "</hazelcast-client>";
+
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
```

Comment:
Perhaps pull out a method like you did testing the config?

```
 private Config buildConfig(String xml) {
        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
        XmlConfigBuilder configBuilder = new XmlConfigBuilder(bis);
        return configBuilder.build();
    }
```

When you are copy pasting code; then your spider senses should start tingling.